crates/cache/src/utils.rs

```diff
@@ -91,6 +90,30 @@ fn collect_table_names(plan: &LogicalPlan, table_names: &mut HashSet<String>) {
     });
 }
 
+#[must_use]
+pub fn is_cache_allowed_for_query(plan: &LogicalPlan) -> bool {
```

Comment:
I'd keep the old name.

```suggestion
pub fn cache_is_enabled_for_plan(plan: &LogicalPlan) -> bool {
```